controllers/topics-controllers.js

```diff
@@ -1,7 +1,26 @@
-const {selectTopics} = require('../models/topics-models.js')
+const {selectTopics, selectArticleById} = require('../models/topics-models.js')
+const endpointDescriptions = require('../endpoints.json')
 
 exports.getTopics = (req, res, next) => {
     selectTopics().then((topics) => {
         res.status(200).send({ topics })
     })
+}
+
+exports.getEndpoints = (req, res, next) => {
+    res.status(200).send(endpointDescriptions)
+}
+
+exports.getArticleById = (req, res, next) => {
+    const { article_id } = req.params
+    selectArticleById(article_id).then((article) => {
+        if (!article) {
+            next({status: 404, msg: 'article does not exist'})
```

Comment:
Ideally we want to be using promise.reject here. Take a look what this looks like below:
https://notes.northcoders.com/courses/js-back-end/error-handling#custom-errors

This logic may also be better in your model.